src/project/manifest/feature.rs

```diff
@@ -248,12 +249,45 @@ impl<'de> Deserialize<'de> for Feature {
 
         let inner = FeatureInner::deserialize(deserializer)?;
 
-        let mut dependencies = HashMap::from_iter([(SpecType::Run, inner.dependencies)]);
+        // check duplicate dependencies (run, host, build)
+        let mut dependencies = inner.dependencies.clone();
+        dependencies.extend(inner.host_dependencies.clone().unwrap_or_default());
+        dependencies.extend(inner.build_dependencies.clone().unwrap_or_default());
+        let mut dependency_map = HashSet::new();
+        for package in dependencies.keys() {
+            let package_name = PackageName::from_str(package).map_err(serde::de::Error::custom)?;
```

Comment:
Can't this check be merged with an insert into the dependencies on line 266.  Now we're doing the `from_str` twice.

Reply:
I thought about this as well but I'm afraid it will further complicate the logic due to error handling:

```
261 |             .collect();
    |              ^^^^^^^ the trait `Hash` is not implemented for `InvalidPackageNameError
```

It is kinda painful but I will look into other ways of doing this.

Reply:
You could also put the error on the inserts right?